File: cpp/cpp_1000-/1108.go
// Go
// leetcode 1108
// https://leetcode.cn/problems/defanging-an-ip-address/

func defangIPaddr(address string) string {
	bs := make([]byte, 0, len(address))
	for _, c := range address {
		if c == '.' {
			bs = append(bs, '[', '.', ']')
		} else {
			bs = append(bs, byte(c))
		}
	}
	return string(bs)
}

File: cpp/cpp_1000-/1108.cc
// C++
// leetcode 1108
// https://leetcode.cn/problems/defanging-an-ip-address/

class Solution {
public:
    string defangIPaddr(string address) {
    	vector<char> tmp;
    	for(auto c : address) {
    		if(c != '.') {
    			tmp.push_back(c);
    		} else {
    			tmp.push_back('[');
    			tmp.push_back('.');
    			tmp.push_back(']');
    		}
    	}
    	return string(tmp.begin(), tmp.end());
    }
};